pkg/utils/codec.go

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Intel Corporation
+
+// Package utils contains utility functions
+package utils
+
+import (
+	"google.golang.org/protobuf/proto"
+)
+
+// ProtoCodec encodes/decodes Go values to/from PROTOBUF.
+type ProtoCodec struct{}
+
+// Marshal encodes a Go value to PROTOBUF.
+func (c ProtoCodec) Marshal(v interface{}) ([]byte, error) {
+	return proto.Marshal(v.(proto.Message))
```

Comment:
If it turns out to be not `proto.Message` type, it will panic and this is probably what we need to avoid. Would it better to check and return an error in case of failed type assertion?

Reply:
good catch, fixed